src/main/java/org/jabref/gui/shared/SharedDatabaseLoginDialogView.java

```diff
@@ -96,6 +98,7 @@ private void initialize() {
                 tabContainer,
                 dialogService,
                 preferencesService,
+                aiService,
```

Comment:
Really (https://github.com/JabRef/jabref/pull/11430#discussion_r1667343717).

That's why I initialy didn't want to introduce a new `EmbeddingsGenerationTask` alongside the PDF indexing

Reply:
The issue is more a JabRef design issue. That a) the dialog triggers the opening, too (IMHO too much logic) and b) the dependencies are passed through as parameters and not by some injector (which is debatable - and we currently vote for passing it all along)

Reply:
@calixtus if you have time, look at this. I think, we need to keep as is, because the login dialog also opens the database. And that thing needs the ai service.

Reply:
Yes, koppor is right, the dialog should not be responsible for opening the database, but the frame, depending on the result of the dialog. (because a dialog is - as the name says - a question to the user). I think the right way would be to do the logic in a SimpleCommand, that also opens the LoginDialog and then uses a supplied (Supplier!) command to perform.

Buuuut for now, this is ok. Larger architectural changes are not part of your project, but for the JabCon(?). ;-)